Emerald Insight.js

```diff
@@ -147,18 +147,23 @@ function scrape(doc, url) {
 				item.tags.push(tag.textContent);
 			}
 			// authors are in RIS as lastname firstname(s), though not necessarily correctly so
-			// trying to correct for this
+			// trying to correct for this. Editors are firstname lastname
 			for (let i = 0; i < item.creators.length; i++) {
-				if (!item.creators[i].firstName && item.creators[i].lastName.includes(" ")) {
-					var author = doc.querySelector("a[href*='" + item.creators[i].lastName.split(" ")[0] + "'][class='contrib-search']");
-					if(author) {
+				lastName = item.creators[i].lastName.split(" ");
+				if (!item.creators[i].firstName && lastName) {
```

Comment:
I'm not sure I understand this part. What is this condition meant to test? Just if there's no space in the RIS name? I was suggesting that we always use the author spans when the elements are present and only fall back to splitting the RIS authors when no spans are available.

Reply:
Originally the regex was tested on this and it needed a space to return something, I didn't threw it away in case there is nothing in the RIS data.

> I was suggesting that we always use the author spans when the elements are present and only fall back to splitting the RIS authors when no spans are available

I think that's what it does, the spans are wrapped in the `a` tag that is queried on the next line. If nothing is returned we fall back to a split of the RIS data (different for authors and editors as per @zuphilip remark)

Reply:
I think what Dan suggest here is something like this code:
```Js
var authorsNodes = doc.querySelectorAll("a.contrib-search");
if (authorsNodes.length > 0) {
   // prefer the authors information from the website as it contains the last and first name separately
   // where the RIS data does not separate them correctly (it uses a space instead of comma)
   // but the editors are only part of the RIS data
   var authors = [];
   for (let author of authorsNodes) {
      authors.push({
         firstName: text(author, "span.given-names"),
         lastName: text(author, "span.surname"),
         creatorType: "author"
      });
   }
   var otherContributors = item.creators.filter(creator => creator.creatorType !== "author");
   item.creators = authors.concat(separateNames(otherContributors));
}
else {
   Z.debug("No span tag for " + item.creators[i].lastName);
   item.creators = separateNames(item.creators);
}
```

Note: this codes is untested and you would then need to write a new function `separateNames` by copying some of the lines of the current code.

Reply:
Thanks a lot @zuphilip I was a bit lost on that one